### ISSUE TITLE
Mastodon (official app / iOS) does not connect

### ISSUE THREAD

**Issue by boerge30** [superseriousbusiness/gotosocial#559]
I can't connect to my GoToSocial installation (Docker v0.3.3) with the official Mastodon iOS app.

* "My" GoToSocial will be found in the login window
* Login/Password --> Allow --> Return to login window

It works great with the Android Mastodon app (official).

Here are some Portainer logs (maybe these is helpful):

`time=2022-05-13T07:07:27Z level=info msg=[5.086661ms] OK: wrote 767 bytes clientIP=172.20.0.1 latency=5.086661ms method=GET path=/oauth/authorize statusCode=200 userAgent=Mozilla/5.0 (iPhone; CPU iPhone OS 15_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Mobile/15E148 Safari/604.1
time=2022-05-13T07:07:33Z level=info msg=[10.378251ms] Found: wrote 23 bytes clientIP=172.20.0.1 latency=10.378251ms method=POST path=/oauth/authorize statusCode=302 userAgent=Mozilla/5.0 (iPhone; CPU iPhone OS 15_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Mobile/15E148 Safari/604.1
time=2022-05-13T07:07:33Z level=info msg=[18.590368ms] OK: wrote 152 bytes clientIP=172.20.0.1 latency=18.590368ms method=POST path=/oauth/token statusCode=200 userAgent=Mastodon/121 CFNetwork/1331.0.7 Darwin/21.4.0
time=2022-05-13T07:07:27Z level=info msg=[90.117113ms] Found: wrote 23 bytes clientIP=172.20.0.1 latency=90.117113ms method=POST path=/auth/sign_in statusCode=302 userAgent=Mozilla/5.0 (iPhone; CPU iPhone OS 15_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Mobile/15E148 Safari/604.1
time=2022-05-13T07:07:11Z level=info msg=[1.39466ms] OK: wrote 708 bytes clientIP=172.20.0.1 latency=1.39466ms method=GET path=/auth/sign_in statusCode=200 userAgent=Mozilla/5.0 (iPhone; CPU iPhone OS 15_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Mobile/15E148 Safari/604.1`

**Comment by tsmethurst** [comment]
Heya! Thanks for opening this. Unfortunately none of us in the dev team have iphones so it's really hard to test this :upside_down_face: Is there a particular error message that the ios app gives once it returns from the login flow?

**Comment by boerge30** [comment]
No, unfortunately not. It jumps to the beginning of the login process without a message.

**Comment by carbontwelve** [comment]
I tailed my log and attempted to connect with the mastodon app, these two log items are all that came out of it. The app doesn't even get to the login stage, it makes these two requests and then shows a spinner:
![image](https://user-images.githubusercontent.com/464699/169326088-35b3adf8-055f-407a-860a-4d023e7c96c1.png)

```
2022-05-19T14:45:27.149207944Z app[web.1]: time=2022-05-19T14:45:27Z level=info msg=[369.362µs] Bad Request: wrote 66 bytes clientIP=172.17.0.1 latency=369.362µs method=GET path=/.well-known/webfinger statusCode=400 userAgent=Mastodon/121 CFNetwork/1331.0.7 Darwin/21.4.0
2022-05-19T14:45:27.223511506Z app[web.1]: time=2022-05-19T14:45:27Z level=info msg=[2.720214ms] OK: wrote 380 bytes clientIP=172.17.0.1 latency=2.720214ms method=GET path=/api/v1/instance statusCode=200 userAgent=Mastodon/121 CFNetwork/1331.0.7 Darwin/21.4.0
```

This could be due to the GET request to `/.well-known/webfinger` responding with a 400, alternatively there could be something missing in the `/api/v1/instance` response that the app is expecting.

**Comment by tsmethurst** [comment]
Thanks for doing this investigation! it looks like it's probably related to #345

**Comment by tsmethurst** [comment]
I'm not sure exactly what we changed to get this working, but apparently the iOS app works for GoToSocial (0.3.5 upwards), so I'm gonna close this issue :)

**Comment by tsal** [comment]
This is still an issue with other clients, not just the official one, on multiple platforms, but primarily it seems like macOS is getting hit the hardest, possibly due to most frameworks targeting the platform are written using strict JSON marshaling. an empty value is a busted value if it's required by the protocol.

**Comment by tsal** [comment]
@carbontwelve 's response got me digging:

https://github.com/mastodon/mastodon-ios/blob/main/MastodonSDK/Sources/MastodonSDK/Entity/Mastodon%2BEntity%2BInstance.swift

so.. my previous comment will have a strike through. This was probably a bug in the Mastodon client, since the field in the SDK is nil-able.  It's possible these clients are handling a nil value poorly. however, I did open another issue, #671 as it seems like we should be able to set this setting, since we're returning an empty (instead of null) value.
